### PR TITLE
Fixing issue in ResolveBackendConfigs

### PR DESCRIPTION
**Pull request by oandreeva-nv**
This PR fixes [issue](https://github.com/triton-inference-server/server/issues/5277) when not all backend configurations were properly resolved.

### CODE REVIEW

**Review by nv-kmcgill53 — DISMISSED** [review]

**Review by GuanLuo — APPROVED** [review]
I think the resolution logic can be simplified to use map as local variable to resolve the "specific setting overrides global setting":
```c++
map<string, string> lconfig;
for (setting : global) {
    lconfig[setting.first] = setting.second;
}
for (setting : specific) {
    lconfig[setting.first] = setting.second;
}
for (map_item : lconfig) {
    config.emplace_back(map_item);
}
...
```

File: src/backend_model.cc [hunk 1]
```diff
@@ -305,15 +305,15 @@ TritonModel::ResolveBackendConfigs(
     }
 
     // add the rest of the global configs
-    if (global_index < global_backend_config.size()) {
-      auto& current_global_setting = global_backend_config.at(global_index);
+    while (global_index < global_backend_config.size()) {
```
Comment (GuanLuo) [inline src/backend_model.cc:308]: use `for (;global_index < global_backend_config.size(); ++global_index)` for clarity, same for below

File: src/backend_model.cc [hunk 2]
```diff
@@ -276,45 +277,19 @@ TritonModel::ResolveBackendConfigs(
...
-          0) {
-        config.push_back(current_specific_setting);
-        ++specific_index;
-      } else {
-        config.push_back(current_global_setting);
-        ++global_index;
-      }
+    std::map<std::string, std::string> lconfig;
+    // Accumulate all global settings
+    for (auto& setting : global_backend_config){
```
Comment (GuanLuo) [inline src/backend_model.cc:282]: You can get rid of line 262-279 based on the cases already captured in your new changes
```suggestion
    if (global_itr != backend_cmdline_config_map.end()) {
      // Accumulate all global settings
      for (auto& setting : global_itr->second){
      ...
```

Comment (oandreeva-nv) [reply src/backend_model.cc:282]: that is true, missed that. Thank you

**Review by GuanLuo — APPROVED** [review]

**Pull request merged**